### PR TITLE
Feature/variance align control

### LINKED ISSUE #113: support "alignment" for variance data set
## Use Case

variance data set single cell images have all been aligned, meaning they all have a translation in xy and a rotation about z. 
We want to allow users of CFE to be able to see the 3d cells both aligned and unaligned.

## Solution

We need to forward this data from the cell-feature-data through CFE and into the 3d viewer. 
3d viewer will get some kind of button to toggle between aligned and unaligned (in general, apply transform/remove transform).  The base volume-viewer already can accept a transform and has an api to toggle it.   
We just need to marshall the data and convert it into the form that the viewer wants.

**_Key question_**:  What is the top-level data spec (ie what will `cell-feature-data` receive?). Does a per-image transform belong with the feature data?  Is there some new data structure we should use instead?   If we can nail this down then @vianamp can have the right guideline of how to format the data.

Initial proposal was to just use features and supply specially named features like `translate_x`, `translate_y` (in pixels?), and `rotate_z` (radians or degrees).  However this feels slightly like an abuse of the features columns if they will never be plotted in the scatterplot, and could affect performance on this large of a dataset.

### PR DESCRIPTION
**Pull request by frasercl**
Problem
=======
#113: "Variance data set single cell images have all been aligned, meaning they all have a translation in xy and a rotation about z.
We want to allow users of CFE to be able to see the 3d cells both aligned and unaligned."

Solution
========
Resolve #113: We've previously decided on a format to add transform data. Add new alignment control component, (somewhat hacky) code to inject it into the viewer toolbar, and associated redux plumbing to read transforms and manage aligned/unaligned state.

## Type of change
* New feature (non-breaking change which adds functionality)
* This change requires updated or new tests

Steps to Verify:
----------------
1. Run `npm install` to ensure the viewer is updated.
2. Load a dataset without any transform metadata (i.e. with `null` or nothing at all in the new `transform` field of `FileInfo`). Alignment control should not appear at all.
3. Load a dataset with transform metadata. Align controls should appear, with an info tooltip and On/Off radio buttons.
4. Click "On." The specified transform should be applied.

Screenshots (optional):
-----------------------
image
image
(The screenshots above are running on a fake transform injected via redux devtools - this is not necessarily the actual alignment of this cell!)

### CODE REVIEW

File: src/components/AlignControl/index.tsx
Comment (toloudis) [inline src/components/AlignControl/index.tsx:11]: I wish this string were data. It kinda has no place in this app's source code.  In fact it would be cool if this type of value were encapsulated in a file called "hacksBecauseWeCantMakeThisAppDataDriven.ts" (not literally, but I think there are still a small amount of things like this sprinkled around).  

Maybe a file like extraData.ts containing misc data that could be per dataset:
```
"hipsc_v2022.1":   { "alignInfoText":"...bla bla bla", otherData2: foo, otherData3: bar }
```
(Megan's PR also has some data that would fit in there)

File: src/components/AlignControl/index.tsx
Comment (toloudis) [inline src/components/AlignControl/index.tsx:19]: I'm ok approving this as is but I still wish this were just a "Apply Transform" button in the viewer toolbar that was only being restyled and relabeled by CFE.

File: src/containers/Cfe/index.tsx
Comment (toloudis) [inline src/containers/Cfe/index.tsx:247]: this seems like a very high level of the app to be adding this functionality.  I know it works just fine like this but I wonder if there is a way that would keep its implementation closer to the component where it's associated.

**Review by toloudis — APPROVED** [review]
I'm ok to merge this if my suggestions could be saved for another pass at this code some day.

File: src/containers/Cfe/index.tsx
Comment (frasercl) [reply src/containers/Cfe/index.tsx:247]: The docs advise that components may be responsible for their own state if the value:

> 1. is absolutely of no interest to the remainder of the application and never will be;
> 2. is not necessary for "hydrating" the application on initial load;
> 3. and cannot be derived from state that exists in the Redux store.

I buy that this case meets all these conditions, the only one I find remotely uncertain is 1: would there ever be a case where another component cares whether alignment is active?

File: src/components/AlignControl/index.tsx
Comment (frasercl) [reply src/components/AlignControl/index.tsx:19]: This is certainly quite possible and would replace the hack I used to get the control into the toolbar with a marginally more limited and elegant hack, but would come at the cost of a bit of flexibility. In particular, I'm not certain there's a great way to add or change tooltip text without owning the component, so we might have to sacrifice that small bit of the specification. Personally, I could go either way: I think it would be nicer for the viewer to own this control, but can be persuaded that since so much of the specification is specific to this limited use case, it makes sense for it all to be defined at this level. I'd also be fine with merging this as-is now and setting an issue on the tracker to investigate moving responsibility down.

Comment (toloudis) [reply src/components/AlignControl/index.tsx:19]: At the risk of sounding ridiculous, props, and/or styles, are one way to replace the tooltip and relayout things inside a child component. (ridiculous because it scales out to allowing every button and label to be restyled) But yeah I wouldn't derail this PR for that kind of change unless it seemed like something would break.  Future work if at all...

File: src/containers/Cfe/index.tsx
Comment (frasercl) [reply src/containers/Cfe/index.tsx:247]: Looking closer at the code, I remember why I did it this way: props passed in to the viewer are computed by a Redux selector (`getPropsForVolumeViewer`) which is unaware of state stored in `Cfe`. It's possible to break this convention and pass extra props in from the component (I already broke the ice with this in an earlier PR by passing the viewer a handler for when the control panel closes). But given that this bit of state is used to derive a prop (`transform`, either zeroed out or populated from `FileInfo` depending on the value of `alignActive`) rather than being immediately passed as a prop itself, it felt more idiomatic to do that derivation in the selector.

File: src/components/AlignControl/index.tsx
Comment (frasercl) [reply src/components/AlignControl/index.tsx:11]: Seems appropriate to defer this as well until both PRs are merged.

**Pull request merged**
